pkg/envoy/lds/inmesh.go

```diff
@@ -248,6 +272,45 @@ func (lb *listenerBuilder) getInboundTCPFilters(trafficMatch *trafficpolicy.Traf
 	return filters, nil
 }
 
+func buildTCPLocalRateLimitFilter(config *policyv1alpha1.TCPLocalRateLimitSpec, statPrefix string) (*xds_listener.Filter, error) {
+	if config == nil {
+		return nil, nil
+	}
+
+	var fillInterval time.Duration
+	switch config.Unit {
+	case "second":
+		fillInterval = time.Second
+	case "minute":
+		fillInterval = time.Minute
+	case "hour":
+		fillInterval = time.Hour
+	default:
+		return nil, errors.Errorf("invalid unit %q for TCP connection rate limiting", config.Unit)
+	}
+
+	rateLimit := &xds_local_ratelimit.LocalRateLimit{
+		StatPrefix: statPrefix,
+		TokenBucket: &xds_type.TokenBucket{
+			MaxTokens:     config.Connections + config.Burst,
+			TokensPerFill: &wrapperspb.UInt32Value{Value: config.Connections},
+			FillInterval:  durationpb.New(fillInterval),
```

Comment:
The API is designed such that users can specify the throttling per unit of time, and seconds, minute, hours are going to be the most commonly used. Extending this to let's say milliseconds is going to be straightforward. We don't want to expose the duration directly because it's more intuitive for users to configure rate limiting as #connections/requests per unit of time, vs needing to reason about how a token bucket works. A similar approach has been taken by other solutions, see [this](https://docs.google.com/document/d/1VTdte9v0JpX7odMtXlGsLlGwDlnXBAEZAf8PMJ_EyY4/edit#heading=h.g9ti13njgadp).